From 1aa4b3d83524b1e545eeed01c5f9a9b4dbcf83bc Mon Sep 17 00:00:00 2001
From: "Matthew Wilcox (Oracle)" <willy@infradead.org>
Date: Mon, 9 Jan 2023 05:18:22 +0000
Subject: [PATCH 10/11] mm: Remove filemap_fdatawait_range_keep_errors()

This function is now the same as filemap_fdatawait_range(), so change
both callers to use that instead.

Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>
---
 fs/jbd2/commit.c        | 12 +++++-------
 include/linux/pagemap.h |  2 --
 2 files changed, 5 insertions(+), 9 deletions(-)

diff --git a/fs/jbd2/commit.c b/fs/jbd2/commit.c
index 4810438b7..36aa1b117 100644
--- a/fs/jbd2/commit.c
+++ b/fs/jbd2/commit.c
@@ -221,11 +221,10 @@ EXPORT_SYMBOL(jbd2_submit_inode_data);
 int jbd2_wait_inode_data(journal_t *journal, struct jbd2_inode *jinode)
 {
 	if (!jinode || !(jinode->i_flags & JI_WAIT_DATA) ||
-		!jinode->i_vfs_inode || !jinode->i_vfs_inode->i_mapping)
+	    !jinode->i_vfs_inode || !jinode->i_vfs_inode->i_mapping)
 		return 0;
-	return filemap_fdatawait_range_keep_errors(
-		jinode->i_vfs_inode->i_mapping, jinode->i_dirty_start,
-		jinode->i_dirty_end);
+	return filemap_fdatawait_range(jinode->i_vfs_inode->i_mapping,
+				jinode->i_dirty_start, jinode->i_dirty_end);
 }
 EXPORT_SYMBOL(jbd2_wait_inode_data);
 
@@ -270,9 +269,8 @@ int jbd2_journal_finish_inode_data_buffers(struct jbd2_inode *jinode)
 {
 	struct address_space *mapping = jinode->i_vfs_inode->i_mapping;
 
-	return filemap_fdatawait_range_keep_errors(mapping,
-						   jinode->i_dirty_start,
-						   jinode->i_dirty_end);
+	return filemap_fdatawait_range(mapping, jinode->i_dirty_start,
+			jinode->i_dirty_end);
 }
 
 /*
diff --git a/include/linux/pagemap.h b/include/linux/pagemap.h
index 573b8cce3..7fe2a5ec1 100644
--- a/include/linux/pagemap.h
+++ b/include/linux/pagemap.h
@@ -40,8 +40,6 @@ static inline int filemap_fdatawait(struct address_space *mapping)
 	return filemap_fdatawait_range(mapping, 0, LLONG_MAX);
 }
 
-#define filemap_fdatawait_range_keep_errors(mapping, start, end)	\
-	filemap_fdatawait_range(mapping, start, end)
 #define filemap_fdatawait_keep_errors(mapping)	filemap_fdatawait(mapping)
 
 bool filemap_range_has_page(struct address_space *, loff_t lstart, loff_t lend);
-- 
2.39.0.rc2.1.gbd5df96b79

